tests/rbd_mirror/test_rbd_group_mirror_consistency.py

```diff
@@ -359,31 +729,31 @@ def run(**kw):
                 secondary_cluster = val.get("cluster")
 
         pool_types = list(mirror_obj.values())[0].get("pool_types")
-        test_mirror_group_consistency(
-            rbd_primary,
-            rbd_secondary,
-            client_primary,
-            client_secondary,
-            primary_cluster,
-            secondary_cluster,
-            pool_types,
-            **kw,
-        )
-        log.info(
-            "Test verifying group level data consistency on secondary after primary site disaster passed"
-        )
+        test_func = kw["config"]["operation"]
+        if test_func in test_map:
+            test_map[test_func](
+                rbd_primary,
+                rbd_secondary,
+                client_primary,
+                client_secondary,
+                primary_cluster,
+                secondary_cluster,
+                pool_types,
+                **kw,
+            )
+
+        log.info("Test %s passed", test_name)
     except Exception as e:
-        log.error(
-            "Test verifying group level data consistency on secondary after primary site disaster failed: "
-            + str(e)
-        )
+        log.error("Test %s failed with error %s", test_name, e)
         return 1
 
     finally:
-        exec_cmd(
-            node=client_secondary,
-            cmd="ceph orch start rbd-mirror",
-        )
+        if "disaster" in test_name:
+            exec_cmd(
+                node=client_secondary,
```

Comment:
if the disaster from the primary test case it fails, I don't think it needs to be in the finally block it should be handled in the test module so that all primary and secondary disasters tests can be handled.